indexer/osm_editor.cpp

```diff
@@ -678,21 +690,31 @@ NewFeatureCategories Editor::GetNewFeatureCategories() const
   return res;
 }
 
-namespace
+FeatureID Editor::GenerateNewFeatureId(MwmSet::MwmId const & id)
 {
-FeatureID GenerateNewFeatureId(FeatureID const & oldFeatureId)
-{
-  // TODO(AlexZ): Stable & unique features ID generation.
-  // TODO(vng): Looks like new feature indexes should uninterruptedly continue after existing indexes in mwm file.
-  static uint32_t newIndex = 0x0effffff;
-  return FeatureID(oldFeatureId.m_mwmId, newIndex++);
+  // TODO(vng): Double-check if new feature indexes should uninterruptedly continue after existing indexes in mwm file.
+  uint32_t featureIndex = kStartIndexForCreatedFeatures;
+  auto const found = m_features.find(id);
+  if (found != m_features.end())
+  {
+    // Scan all already created features and choose next available ID.
+    for (auto const & feature : found->second)
+    {
+      if (feature.second.m_status == FeatureStatus::Created && featureIndex <= feature.first)
+        featureIndex = feature.first + 1;
```

Comment:
The issue is that instead of selection among two quite natural approaches (select the smallest free id and global monotonically increasing counter) you have implemented quite unnatural approach.